### ISSUE TITLE
Thread for unzipping does not exit when run in another thread

### ISSUE THREAD

**Issue by Cyb0rk** [srikanth-lingala/zip4j#204]
If the option `setRunInThread(true)` is set, the thread for unzipping wont't exit if it is run in another thread.

Example:

```
import net.lingala.zip4j.ZipFile;
import net.lingala.zip4j.exception.ZipException;

public class Main {
    public static void main(String[] args) {
        System.out.println("Application started");

        Thread thread = new Thread(() -> {
            String testZipFile = "S:\\test\\test.zip";
            ZipFile zipFile = new ZipFile(testZipFile);
            zipFile.setRunInThread(true);


            try {
                // This thread won't exit if run in another thread
                zipFile.extractAll("S:\\test\\");
            } catch (ZipException e) {
                System.err.println("Error extracting file: " + testZipFile);
                e.printStackTrace();
            }
        });

        thread.start();
    }
}

```

Is this working as intended? Currently I have to stop it manually with `getExecutorService().shutdown();`.

**Comment by srikanth-lingala** [comment]
When I tried your code above, the application and the thread seems to have exited. How did you verify that the thread is still running? I tested with v2.6.1 and on a Mac (although I guess OS shouldn't really matter here).

**Comment by Cyb0rk** [comment]
I just reproduced it with 2.6.1 on Windows 10.

I verify it because the application doesn't exit correctly and runs forever. Here is a screenshot after running over seven minutes and the image is already extracted after a second from the test zip file. If I use it without threading it will end immediately after unzipping.

If you like I can build you a .jar file with parameters for the zip input and output file. Then you can reproduce it with the same code as I do (for this test, I used the excact code sample from above).

![image](https://user-images.githubusercontent.com/95137/88848718-26fe3300-d1e9-11ea-8a18-da9d6fbb2b84.png)

**Comment by srikanth-lingala** [comment]
Yes, can you please share your code? Because with the same code from your screenshot, the program exits normally for me.

**Comment by Cyb0rk** [comment]
> Yes, can you please share your code? Because with the same code from your screenshot, the program exits normally for me.

Of course. See attached Test-1.0.jar, just run it via `java -jar Test-1.0.jar`. Create a ZIP File at `C:\test\test.zip` and it will be extracted to `C:\test\test.zip`. The source is the same as in the screenshot except for the drive `S:\` changed to `C:\`.

[Test-1.0.zip](https://github.com/srikanth-lingala/zip4j/files/5105053/Test-1.0.zip)

**Comment by srikanth-lingala** [comment]
Issue fixed in v2.6.2 released today.
